via/static/scripts/video_player/hooks/test/use-app-layout-test.js

```diff
@@ -0,0 +1,82 @@
+import { render } from 'preact';
+import { useRef } from 'preact/hooks';
+import { act } from 'preact/test-utils';
+
+import { waitFor } from '../../../test-util/wait';
+import { useAppLayout } from '../use-app-layout';
+
+describe('useAppLayout', () => {
+  let container;
+  let appSize;
+
+  function findElementByTestId(testId) {
+    return container.querySelector(`[data-testid=${testId}]`);
+  }
+
+  // Create a fake component to mount in tests that uses the hook
+  function FakeComponent() {
+    const myRef = useRef();
+    appSize = useAppLayout(myRef);
+    return (
+      <div
+        ref={myRef}
+        data-app-size={appSize}
+        style="width:100%"
+        data-testid="appContainer"
+      >
+        <button>Hi</button>
+      </div>
+    );
+  }
+
+  function renderComponent() {
+    act(() => {
+      render(<FakeComponent />, container);
```

Comment:
NB: Using `preact`'s `render` here. I could not for the life of me find a test structure that worked with enzyme's `mount`.